src/backend/distributed/executor/repartition_join_execution.c

```diff
@@ -71,6 +75,20 @@ ExecuteDependentTasks(List *topLevelTasks, Job *topLevelJob)
 }
 
 
+/*
+ * EnsureNoVisibilityProblem makes sure that the tasks won't have
+ * any visibility problems.
+ */
+static void
+EnsureNoVisibilityProblem(List *taskList)
+{
+	if (TransactionAccessedLocalPlacement && AnyTaskAccessesLocalNode(taskList))
```

Comment:
checking TransactionAccessedLocalPlacement  is redundant, since it happens in ErrorIfTransactionAccessedPlacementsLocally